### ISSUE TITLE
Stuck at train.fit() and print nothing

### ISSUE THREAD

**Issue by Longyichen** [mosaicml/composer#2733]
** Environment **
Collecting system information...
---------------------------------
System Environment Report        
Created: 2023-11-20 20:47:01 CST
---------------------------------

PyTorch information
-------------------
PyTorch version: 2.0.1+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (GCC) 8.2.0
Clang version: 3.8.0 (tags/RELEASE_380/final)
CMake version: version 3.25.0
Libc version: glibc-2.27

Python version: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.10.0-1.0.0.28-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA A100-SXM4-80GB
GPU 1: NVIDIA A100-SXM4-80GB
GPU 2: NVIDIA A100-SXM4-80GB
GPU 3: NVIDIA A100-SXM4-80GB
GPU 4: NVIDIA A100-SXM4-80GB
GPU 5: NVIDIA A100-SXM4-80GB
GPU 6: NVIDIA A100-SXM4-80GB
GPU 7: NVIDIA A100-SXM4-80GB

Nvidia driver version: 525.125.06
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.6.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              160
On-line CPU(s) list: 0-159
Thread(s) per core:  2
Core(s) per socket:  20
Socket(s):           4
NUMA node(s):        4
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz
Stepping:            7
CPU MHz:             3200.097
CPU max MHz:         3900.0000
CPU min MHz:         1000.0000
BogoMIPS:            5000.00
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            1024K
L3 cache:            28160K
NUMA node0 CPU(s):   0-19,80-99
NUMA node1 CPU(s):   20-39,100-119
NUMA node2 CPU(s):   40-59,120-139
NUMA node3 CPU(s):   60-79,140-159
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts pku avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] pytorch-ranger==0.1.1
[pip3] torch==2.0.1+cu118
[pip3] torch-optimizer==0.3.0
[pip3] torchaudio==2.0.2+cu118
[pip3] torchmetrics==1.0.3
[pip3] torchvision==0.15.2+cu118
[conda] numpy                     1.24.1                   pypi_0    pypi
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] torch                     2.0.1+cu118              pypi_0    pypi
[conda] torch-optimizer           0.3.0                    pypi_0    pypi
[conda] torchaudio                2.0.2+cu118              pypi_0    pypi
[conda] torchmetrics              1.0.3                    pypi_0    pypi
[conda] torchvision               0.15.2+cu118             pypi_0    pypi

Composer information
--------------------
Composer version: 0.16.3
Composer commit hash: None
Host processor model name: Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz
Host processor core count: 80
Number of nodes: 1
Accelerator model name: NVIDIA A100-SXM4-80GB
Accelerators per node: 1
CUDA Device Count: 8

** To reproduce

1. Run composer -n 8 $TRAIN_SCRIPT \
2. Exit or interrupt the composer program for various reasons (such as ctrl-c)
3.Restart composer -n 8 $TRAIN_SCRIPT \

The program is stuck at train.fit()
The terminal does not show any errors
The GPU occupies memory but does not occupy UTL, that is, it is not operating normally.

```
above info...
eval_first: false
autoresume: true
dist_timeout: 1800.0
n_gpus: 8
device_train_batch_size: 4
device_train_grad_accum: 4
n_params: 6738773034

Starting training...
******************************
Config:
enabled_algorithms/GradientClipping: true
node_name: unknown because NODENAME environment variable not set
num_gpus_per_node: 8
num_nodes: 1
rank_zero_seed: 17

******************************

stuck here, no any info, no end

```

How to check the errors that occur at this time? I can't solve this error?

## Expected behavior

The program will run normally and print out the information of each epoch

## Additional context

also raise error https://github.com/princeton-nlp/LLM-Shearing/issues/23

**Comment by mvpatel2000** [comment]
Could you retry rerunning this with `python_log_level='debug'`? You can see it as an argument to Trainer in the [docs](https://docs.mosaicml.com/projects/composer/en/latest/api_reference/generated/composer.Trainer.html#trainer). 

Just to clarify, this only happens the _second_ time you run the script?

What are you using for your dataset/dataloader? It could be still spinning the dataloader for resumption if your tool does not support mid-epoch resumption?

**Comment by Longyichen** [comment]
Hi, I tried this setting but it didn't work. 
Since I am using a cloud service server, it may be built based on docker. I am not sure if docker is causing the problem.

Every time I start the environment/enter the server for the first time, this code runs perfectly. But when I interrupt the operation several times, it doesn't work normally.

Although i set autoresume=True, i will alse stuck. 

I'm curious why there is no error message. If the terminal does not display the error message, is there any other place where I can check it? I also wonder if the wrong process has been blocked.

This part of the code can run normally on a single GPU. But there will be problems on multiple GPUs

I process data use Mosaicml's [Streaming](https://github.com/mosaicml/streaming) package, i.e.
from streaming.base.constant import EPOCH_DATA, EPOCH_SHAPE, RESUME
from streaming.base.dataset import StreamingDataset
from streaming.base.shared import SharedMemory, _get_path
from streaming.base.shuffle import get_shuffle
from streaming.base.stream import Stream
from streaming.base.world import World

```
def build_text_dataloader(cfg: DictConfig, device_batch_size: int, dynamic: bool = False, 
                          set_names: str = None, proportion: List[float] = None) -> DataLoader:
    """Builds a text dataloader.

    Args:
        cfg (DictConfig): Configuration dictionary.
        device_batch_size (int): Batch size for one single device.
        dynamic (bool, optional): Whether to use dynamic streaming dataset to load data from each 
        domain dynamically. Defaults to False.
        set_names (str, optional): Name of the dataset. Defaults to None.
        proportion (List[float], optional): Initial proportion of each domain in the dataset. Defaults to None.

    Returns:
        DataLoader: A PyTorch DataLoader object.
    """
    
    if dynamic:
        dataset = TextDynamicStreamingDataset(local=cfg.dataset.local,
                                              max_seq_len=cfg.dataset.max_seq_len,
                                              batch_size=device_batch_size,
                                              shuffle=cfg.dataset.get(
                                                'shuffle', False),
                                              shuffle_seed=cfg.dataset.get(
                                                'shuffle_seed', 9176),
                                              num_canonical_nodes=cfg.dataset.get(
                                                'num_canonical_nodes', 128),
                                              proportion=proportion,
                                              set_names=set_names,
                                              is_uint16=cfg.dataset.get("is_uint16", False))
    else:
        dataset = TextStreamingDataset(
            local=cfg.dataset.local,
            max_seq_len=cfg.dataset.max_seq_len,
            split=cfg.dataset.get('split', None),
            shuffle=cfg.dataset.get('shuffle', False),
            shuffle_seed=cfg.dataset.get('shuffle_seed', 9176),
            num_canonical_nodes=cfg.dataset.get(
                'num_canonical_nodes', 128),
            batch_size=device_batch_size,
            is_uint16=cfg.dataset.get("is_uint16", False))

    tokenizer = AutoTokenizer.from_pretrained(cfg.dataset.tokenizer_name)
    if isinstance(dataset[0], Mapping) and "set" in dataset[0]:
        COLLATE_FN = DataCollatorForLMWithSetName
        collate_fn = COLLATE_FN(
        set_names=set_names,
        tokenizer=tokenizer,
        mlm=False)
    else:
        COLLATE_FN = transformers.DataCollatorForLanguageModeling
        collate_fn = COLLATE_FN(
            tokenizer=tokenizer,
            mlm=False,
        )
    
    return DataLoader(
        dataset,
        collate_fn=collate_fn,
        batch_size=device_batch_size,
        drop_last=cfg.drop_last,
        num_workers=cfg.num_workers,
        pin_memory=cfg.get('pin_memory', True),
        prefetch_factor=cfg.get('prefetch_factor', 2),
        persistent_workers=cfg.get('persistent_workers', True),
        timeout=cfg.get('timeout', 0),
    )
```

**Comment by Longyichen** [comment]
https://github.com/mosaicml/llm-foundry/issues/436
problem solve by pr above
